src/server/common/forum_user_handler.hpp

```diff
@@ -221,6 +221,40 @@ class fuh : public user_handler
 	 */
 	void db_insert_addon_info(const std::string& instance_version, const std::string& id, const std::string& name, const std::string& type, const std::string& version, bool forum_auth, int topic_id);
 
+	/**
+	 * Inserts into the database for when a player logs in.
+	 * 
+	 * @param username The username of who logged in. The username is converted to lower case when inserting in order to allow index usage when querying.
+	 * @param ip The ip address of who logged in.
+	 */
+	unsigned long long db_insert_login(const std::string& username, const std::string& ip);
+
+	/**
+	 * Updates the database for when a player logs out.
+	 * 
+	 * @param login_id The generated ID that uniquely identifies the row to be updated.
+	 */
+	void db_update_logout(unsigned long long login_id);
+
+	/**
+	 * Searches for all players that logged in using the ip address.
+	 * The '*' wildcard can be used to search for partial ip addresses.
```

Comment:
These functions are called after the wildcard conversion so they take SQL wildcards.

Reply:
Updated.